app-doc/heirloom-doctools/heirloom-doctools-160308.ebuild

```diff
@@ -0,0 +1,87 @@
+EAPI=6
+
+inherit flag-o-matic toolchain-funcs multilib readme-gentoo
+
+DESCRIPTION="Classic Unix documentation tools ported from OpenSolaris"
+HOMEPAGE="https://n-t-roff.github.io/heirloom/doctools.html"
+SRC_URI="https://github.com/n-t-roff/heirloom-doctools/releases/download/${PV}/${P}.tar.bz2"
+
+LICENSE="CDDL"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="cxx"
+
+RDEPEND="!sys-apps/groff"
+DEPEND="sys-devel/flex
+	virtual/yacc"
+
+src_prepare() {
+	# mpm uses C++, we'll build it explicitly if we really want to
+	sed -i -e 's:mpm:$(MPM):' makefile || die
+
+	# Fixing compilation issues
+	# specific for versions =<160308, fixed in -git
+	sed -i -e 's:e.h:e.h y.tab.h:' eqn/eqn.d/Makefile.mk || die
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/eqn.d/Makefile.mk || die
+
+	sed -i -e 's:e.h:e.h y.tab.h:' eqn/neqn.d/Makefile.mk || die
+	sed -i -e 's:y.tab.h y.tab.h:y.tab.h:' eqn/neqn.d/Makefile.mk || die
+
+	eapply_user
+}
+
+src_configure() {
+	append-cppflags -D_GNU_SOURCE
+
+	sed \
+		-e "s:@CFLAGS@:${CFLAGS}:" \
+		-e "s:@CXXFLAGS@:${CXXFLAGS}:" \
+		-e "s:@CPPFLAGS@:${CPPFLAGS}:" \
+		-e "s:@LDFLAGS@:${LDFLAGS}:" \
+		-e "s:@CC@:$(tc-getCC):" \
+		-e "s:@CXX@:$(tc-getCXX):" \
+		-e "s:@RANLIB@:$(tc-getRANLIB):" \
+		-e "s:@libdir@:$(get_libdir):" \
+		"${FILESDIR}"/${PV}.config \
+		> "${S}"/mk.config
+}
+
+src_compile() {
+	emake $(use cxx && echo MPM=$(usex cxx mpm ''))
+}
+
+src_install() {
+	dodir /usr/share/heirloom/{doc,ref}tools
+
+	# The build system uses the ROOT variable in place of DESTIDR.
+	emake $(use cxx && echo MPM=$(usex cxx mpm '')) ROOT="${D}" install
+
+	dodoc README CHANGES
+
+	# Rename ptx to avoid a collision with coreutils… maybe this
+	# should be made conditional to userland_GNU (somebody got to
+	# check on FreeBSD).
+	mv "${D}"usr/bin/{,hl-}ptx || die
+	mv "${D}"usr/share/man/man1/{,hl-}ptx.1 || die
+
+	mv "${D}"usr/bin/{,hl-}col || die
+
+	# Rename ta to avoid a collision with app-cdr/pxlinux
+	mv "${D}"usr/bin/{,hl-}ta || die
+
+}
```

Comment:
You should also include the bit of code for installing README.gentoo file, so users can re-read it after installing.